src/goto-programs/goto_coverage.cpp

```diff
@@ -501,6 +540,98 @@ exprt goto_coveraget::gen_not_expr(const exprt &expr, const locationt &loc)
   return not_expr;
 }
 
+expr2tc goto_coveraget::gen_not_expr(const expr2tc &guard)
+{
+  exprt _guard = migrate_expr_back(guard);
+  exprt not_guard = gen_not_expr(_guard, _guard.location());
+  expr2tc _guard2;
+  migrate_expr(not_guard, _guard2);
+  return _guard2;
+}
+
+/*
+  for adding assertions
+  convert
+     assert(0);
+  1: IF(...)
+  to
+  1: assert(0);
+     IF(...)
+*/
+void goto_coveraget::update_goto_target(
```

Comment:
I think you can avoid the label issue if you use the insert_swap instead of insert in `insert_assertion` function. Similar to what is done here: https://github.com/esbmc/esbmc/blob/master/src/goto-programs/abstract-interpretation/interval_analysis.cpp#L123-L128

Could you please check?

Reply:
Sure. Will check. Thanks for the suggestion!